### ISSUE TITLE
Some tests use sample data

### ISSUE THREAD

**Issue by pelson** [SciTools/iris#1598]
All are using ```GloSea4/ensemble_001.pp```:

e.g.
```
======================================================================
ERROR: test_4d_cube_2_dims (iris.tests.analysis.test_stats.Test_corr)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "iris/tests/analysis/test_stats.py", line 38, in setUp
    'ensemble_001.pp'))
  File "/iris/__init__.py", line 327, in load_cube
    cubes = _load_collection(uris, constraints, callback).merged().cubes()
  File "lib/iris/__init__.py", line 267, in _load_collection
    result = iris.cube._CubeFilterCollection.from_cubes(cubes, constraints)
  File "lib/iris/cube.py", line 139, in from_cubes
    for cube in cubes:
  File "lib/iris/__init__.py", line 254, in _generate_cubes
    for cube in iris.io.load_files(part_names, callback, constraints):
  File "lib/iris/io/__init__.py", line 183, in load_files
    all_file_paths = expand_filespecs(filenames)
  File "lib/iris/io/__init__.py", line 166, in expand_filespecs
    for pattern, expanded in glob_expanded.iteritems()])
IOError: One or more of the files specified did not exist ['iris/sample_data/GloSea4/ensemble_001.pp expanded to empty'].
```

**Comment by QuLogic** [comment]
Here are a few more from `master`:
```
======================================================================
ERROR: test_lazy_preserved_save (iris.tests.integration.test_netcdf.TestLazySave)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/tests/integration/test_netcdf.py", line 170, in test_lazy_preserved_save
    acube = iris.load_cube(fpath)
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/__init__.py", line 335, in load_cube
    cubes = _load_collection(uris, constraints, callback).merged().cubes()
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/__init__.py", line 275, in _load_collection
    result = iris.cube._CubeFilterCollection.from_cubes(cubes, constraints)
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/cube.py", line 141, in from_cubes
    for cube in cubes:
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/__init__.py", line 262, in _generate_cubes
    for cube in iris.io.load_files(part_names, callback, constraints):
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/io/__init__.py", line 185, in load_files
    all_file_paths = expand_filespecs(filenames)
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/io/__init__.py", line 168, in expand_filespecs
    for pattern, expanded in glob_expanded.items()])
IOError: One or more of the files specified did not exist ['.../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/test_data/NetCDF/label_and_climate/small_FC_167_mon_19601101.nc expanded to empty'].

======================================================================
ERROR: test (iris.tests.integration.test_pickle.TestGribMessage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/tests/integration/test_pickle.py", line 37, in test
    message = next(messages)
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/fileformats/grib/_message.py", line 55, in messages_from_filename
    with open(filename, 'rb') as grib_fh:
IOError: [Errno 2] No such file or directory: '.../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/test_data/GRIB/fp_units/hours.grib2'

======================================================================
ERROR: test_data (iris.tests.integration.test_pickle.TestGribMessage)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/tests/integration/test_pickle.py", line 46, in test_data
    message = next(messages)
  File ".../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/fileformats/grib/_message.py", line 55, in messages_from_filename
    with open(filename, 'rb') as grib_fh:
IOError: [Errno 2] No such file or directory: '.../lib/python2.7/site-packages/Iris-1.9.0.dev0-py2.7.egg/iris/test_data/GRIB/fp_units/hours.grib2'
```

**Comment by ajdawson** [comment]
@QuLogic - The files you listed are all in the [iris-test-data](https://github.com/SciTools/iris-test-data) repository... This ticket refers to tests using files from the [iris-sample-data](https://github.com/SciTools/iris-sample-data) repository, which is supposed to be reserved for use in examples in documentation rather than testing.

**Comment by QuLogic** [comment]
Ah, you might be right, but tests using iris-test-data should still have a `@skip_data` on them.

**Comment by ajdawson** [comment]
@QuLogic - See #1693.

**Comment by ajdawson** [comment]
Should be all fixed by #1693 and #1714.
